resharper/src/resharper-unity/CSharp/Feature/Services/QuickFixes/UseExplicitTypeInsteadOfStringQuickFix.cs

```diff
@@ -0,0 +1,138 @@
+                yield return new IntentionAction(action, null, anchor);
+            }
+        }
+
+
+        private class UseExplicitTypeInsteadOfStringAction : BulbActionBase
+        {
+            private readonly ITypeElement myType;
+            private readonly IInvocationExpression myOldInvocation;
+            private readonly string myMethodName;
+            private readonly string myLiteralName;
+
+            public UseExplicitTypeInsteadOfStringAction(ITypeElement type, IInvocationExpression oldInvocation,
+                string methodName, string literalName)
+            {
+                myType = type;
+                myOldInvocation = oldInvocation;
+                myMethodName = methodName;
+                myLiteralName = literalName;
+            }
+
+            protected override Action<ITextControl> ExecutePsiTransaction(ISolution solution,
+                IProgressIndicator progress)
+            {
+                using (WriteLockCookie.Create())
+                {
+                    var factory = CSharpElementFactory.GetInstance(myOldInvocation);
+                    if (myOldInvocation.InvokedExpression is IReferenceExpression referenceExpression)
+                    {
+                        var builder = FactoryArgumentsBuilder.Create();
+
+                        var qualifier = referenceExpression.QualifierExpression;
+                        if (qualifier != null)
+                        {
+                            builder.Argument(qualifier);
+                            builder.Append(".");
+                        }
+
+                        builder.Append(myMethodName);
+                        builder.Append("<");
+                        builder.Append(myType.GetClrName().FullName);
+                        builder.Append(">");
+                        builder.Append("()");
+
+                        var newInvocation = factory.CreateExpression(builder.ToString(), builder.ToArguments());
+                        var result = ModificationUtil.ReplaceChild(myOldInvocation, newInvocation);
+
+                        // beatify generated reference
+                        var resultTextRange = result.GetDocumentRange();
+
+                        var psiServices = solution.GetPsiServices();
+                        if (psiServices.GetPsiFile<CSharpLanguage>(resultTextRange) is ICSharpFile file)
+                        {
+                            file.OptimizeImportsAndRefs(
+                                resultTextRange.CreateRangeMarker(DocumentManager.GetInstance(solution)), false, true,
+                                NullProgressIndicator.Instance);
+                        }
+                    }
+                }
+
+                return null;
+            }
+
+            public override string Text =>
+                $"Change '.{myMethodName}(\"{myLiteralName}\")' to '.{myMethodName}<{myType.GetClrName()}>()'";
```

Comment:
This might generate really long QF text that will unnecessary mention the original code and fully qualified name of the target type.
Wouldn't something like `Use type argument`/`Use type-safe overload` be just as descriptive here?